layers/+source-control/git/README.org

```diff
@@ -59,6 +65,163 @@ file.
+      ;; "GitLab"
+      ;; "gitlab.com"
+      ;; ;; Either "no", or "confirm".
+      ;; "no"
+      )
+   #+end_src
+
+   You can evaluate the above source block to configure SSH fully for working
+   with a Git forge. Ensure to set the variables to their appropriate values.
+
+3. Ensure the upstream URLs for the remote repository are SSH form, according to
+   the documentation for the Git forge you are working with.
+
+   GitHub's documentation says the remote repository URL to use for SSH, rather
+   than HTTPS, is:
+
+   #+begin_example
+   git@github.com:syl20bnr/repository.git
+   #+end_example
+
+   Other Git forges may have a different URL scheme. Check the documentation for
+   the forge you would like to use.
+
+Below is a fully worked example of configuring SSH for use with Magit. You will
+be able to clone, pull, and push as normal.
+
+E.g.: Configure Emacs to use the desktop environment login key-chain, ie tell
+Emacs to acquire SSH keys for remote logins from GNOME or KDE key-chains (or
+"wallets"). This is provided by the Free Desktop standard [[https://www.gnu.org/software/emacs/manual/html_mono/auth.html#Secret-Service-API][Secret Service API]].
+
+1. Add the following to your ~spacemacs/user-config~ function:
+   #+begin_src emacs-lisp
+     (setq auth-sources '(default "secrets:Login"))
+   #+end_src
+
+2. Add the =exampleRSAKey= to the SSH configuration.
+   #+begin_src emacs-lisp
+     ((lambda (forgeUsername
+               sshKeyName
+               (&optional (gitForgeProperName "Github")
+                          (gitForgeHostname   "github.com")
+                          (addKeyToAgent-p    "yes")))
+        "Add a new SSH key to the user's login keyring, and\
+             configure it for a Git forge."
+        nil
+        (append-to-file
+         (concat "\n"
+                 "HOST " gitForgeProperName "\n"
+
+                 ;; The .pub file extension is not necessary; it may cause
+                 ;; issues to include it. See man 5 ssh_config
+                 "     IdentityFile ~/.ssh/" sshKeyName "\n"
+                 "     User " forgeUsername "\n"
+                 "     Hostname " gitForgeHostname "\n"
+                 "     AddKeysToAgent " addKeyToAgent-p)
+         nil
+         "~/.ssh/config"))
+
+      "bryce-carson"
+      "~/ssh/exampleRSAKey"
+      )
+   #+end_src
+
+3. Save and reload the configuration changes.
+
+   ~SPC f e R~
+
+4. Restart Emacs, or restart the systemd service managing the Emacs client.
+
+   ~SPC q R~, or ~SPC f d~ (to delete the client frame) then ~systemctl --user restart emacs~.
+
+5. Ensure the remote repository URL is configured.
+
+   1. Configure the remotes for the current local repository: ~SPC g m M C~.
+
+   2. Select the upstream, such as a fork of another repository, or the origin(al) repository, in the minibuffer.
+
+   3. Set the URL to the appropriate scheme for the forge (=url= & =pushurl=): ~git@github.com/syl20bnr/spacemacs.git~.
+      
```

Comment:
Perhaps this section could be simplified to the following (unless I have misunderstood something)
```org
**** Magit Forge with SSH Key
An SSH key can be used instead of a personal access token to 

Set =auth-sources= variable to ="secrets:Login"= to use a login key-ring, unlocking
the key on login to the desktop environment.

#+begin_src emacs-lisp
   (setq auth-sources '(default "secrets:Login"))
#+end_src

Alternative, use ="secrets:session"= to open the key ring only when Emacs is running.

Add SSH configuration to =~/.ssh/config= 

   #+begin_src 
  HOST hostname              # e.g. GitHub / GitLab
  IdentityFile ~/.ssh/rsa_id
  User forgeusername          
  Hostname forge-domain.com  # e.g. github.com
  AddKeysToAgent yes
   #+end_src
```